src/shared/schema/schemas.ts

```diff
@@ -0,0 +1,130 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { mkdirSync, writeFileSync } from 'fs-extra'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import * as filesystemUtilities from '../filesystemUtilities'
+import { getLogger } from '../logger'
+import { CompositeResourceFetcher } from '../resourcefetcher/compositeResourceFetcher'
+import { FileResourceFetcher } from '../resourcefetcher/fileResourceFetcher'
+import { HttpResourceFetcher } from '../resourcefetcher/httpResourceFetcher'
+import { ResourceFetcher } from '../resourcefetcher/resourcefetcher'
+import { normalizeSeparator } from '../utilities/pathUtils'
+import { Schemas } from './schemaService'
+
+const GOFORMATION_MANIFEST_URL = 'https://api.github.com/repos/awslabs/goformation/releases/latest'
+
+export type SchemaType = 'cfn' | 'sam' | 'none'
+
+/**
+ * Loads JSON schemas for CFN and SAM templates.
+ * Checks manifest and downloads new schemas if the manifest version has been bumped.
+ * Uses local, predownloaded version if up-to-date or network call fails
+ * If the user has not previously used the toolkit and cannot pull the manifest, does not provide template autocomplete.
+ * @param extensionContext VSCode extension context
+ */
+export async function getSchemas(extensionContext: vscode.ExtensionContext): Promise<Schemas> {
```

Comment:
The module name, if not this function name, seem a bit generic. Is the plan to extend them beyond CFN schemas? Else I would name it `src/shared/cfnSchemas.ts`, and schemaService can live in the same file. Don't really need an entire namespace (`schemas/`) for this concept.